dspy/primitives/program.py

```diff
@@ -20,26 +21,38 @@ def _base_init(self):
     def __init__(self, callbacks=None):
         self.callbacks = callbacks or []
         self._compiled = False
+        # LM calling history of the module.
+        self.history = []
```

Comment:
Cool! Does this work if there are multiple LMs used in the module for different sub-modules?